### PR TITLE
Fixes #2614 : Uneven Behavior of List Elements

### LINKED ISSUE #2614: Uneven Behavior of List Elements
An issue with ElectronClient.

After adding a few elements in a list item ( NumberList, BulletedList ), if we **press the Enter key**, the **last unused row should get deleted automatically** which doesn't happen in Joplin.<br><br>

Here is how it behaves:<br>
jopnote

Here is how it should behave:<br>
expectnote

### PR DESCRIPTION
**Pull request by mhimanshu712**
Desktop: Added code to detect and remove, empty list elements after the enter key is pressed.

### CODE REVIEW

File: ElectronClient/gui/NoteText.jsx [hunk 1]
```diff
@@ -995,6 +995,23 @@ class NoteTextComponent extends React.Component {
...
+
+				let regexer = /^[\d]+[.] $/;
+				if (line == '- ' || line == '* ' || line == '- [ ] ' || line== '- [x] '|| regexer.test(line)) {
+					let s = that.currentTextOffset();
+
+					const s1 = that.state.note.body.substr(0, s-(line.length+1));
+					const s2 = that.state.note.body.substr(s);
+					console.log(s1);
+					console.log(s2);
+					shared.noteComponent_change(that, 'body', s1 + s2);
```
Comment (laurent22) [inline ElectronClient/gui/NoteText.jsx]: The containing function is called `getNextLineIndent` so it should not have any side-effect, and in fact the note doesn't change so the "change" event should not be triggered. I'm not sure how to best implement what you want to do, but this feels a bit like a hack.

File: ElectronClient/gui/NoteText.jsx [hunk 2]
```diff
@@ -995,6 +995,23 @@ class NoteTextComponent extends React.Component {
...
+
+				// Detects empty List Element
+				// Solves: https://github.com/laurent22/joplin/issues/2614
+
+				let regexer = /^[\d]+[.] $/;
+				if (line == '- ' || line == '* ' || line == '- [ ] ' || line== '- [x] '|| regexer.test(line)) {
+					let s = that.currentTextOffset();
+
+					const s1 = that.state.note.body.substr(0, s-(line.length+1));
+					const s2 = that.state.note.body.substr(s);
```
Comment (Rishgod) [inline ElectronClient/gui/NoteText.jsx]: remove the console.log!

File: ElectronClient/gui/NoteText.jsx [hunk 3]
```diff
@@ -995,6 +995,23 @@ class NoteTextComponent extends React.Component {
...
+					console.log(s1);
```
Comment (Rishgod) [inline ElectronClient/gui/NoteText.jsx]: remove the console.log!

Comment (mhimanshu712) [reply ElectronClient/gui/NoteText.jsx]: Thank You Rish

File: ElectronClient/gui/NoteText.jsx (hunk 1 above)
Comment (mhimanshu712) [reply ElectronClient/gui/NoteText.jsx]: Its kind of hack... But it works!

**Comment by Rishgod** [comment]
@laurent22 I think this is a really neat feature to have, can you please clarify two thing

- could this get merged in its current state with a bit of refactoring

-  if you think the logic is a bit too hacky and unconventional, I can change the logic to this code to something more conventional, just need your confirmation before going ahead with anything.  Thanks!

File: ElectronClient/gui/NoteText.jsx (hunk 1 above)
Comment (mhimanshu712) [reply ElectronClient/gui/NoteText.jsx]: Under the same function ( _getNextLineIndent_ ), we are checking if the next line should start with a List Element character (line 1018-1023). So I thought this was the best place to deal with the issue.
paoa

Comment (mhimanshu712) [reply ElectronClient/gui/NoteText.jsx]: I understand you are worried about the side effect.
I will try something else!
@laurent22

**Comment by tessus** [comment]
@mhimanshu712 why did you close this PR?

**Comment by tessus** [comment]
@Rishgod this PR seems to be abandoned. Feel free to resurrect it....

**Comment by mhimanshu712** [comment]
Sorry @tessus , My exams are going on that's why I could not reply early.
I closed it because I was going to implement it some other way.
My exams will last a few more days, meanwhile if @Rishgod wishes he can work on this!

**Comment by tessus** [comment]
@Rishgod you mentioned you'd be looking into this. is there a status update?

**Comment by Rishgod** [comment]
> @Rishgod you mentioned you'd be looking into this. is there a status update?

I have been busy with other PR's if some else can do it in the meantime that would be great, if no one does it I'll eventually get it done in a week or two

**Pull request closed without merging**